test/test_ControllerOutputRefiner.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+import local_imports
+import rostest
+import unittest
+from controller_output_refiner import ControllerOutputRefiner
+import math
+
+# Do something with local_imports to avoid lint errors
+local_imports.printMessage()
+
+
+class Test_ControllerOutputRefiner(unittest.TestCase):
+
+    # inputSignal is within the bounds
+    def test_saturate_inputWithinBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(math.pi, 2 * math.pi, -2 * math.pi),
+            math.pi
+        )
+
+    # inputSignal is greater than upperBound
+    def test_saturate_greaterThanUpperBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(3.5, 3, -1),
+            3
+        )
+
+    # inputSignal is smaller than lowerBound
+    def test_saturate_smallerThanLowerBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(-15, 10, -10),
+            -10
+        )
+
+    # Corner cases where input signals are close to upper and lower bounds
+    def test_saturate_justAboveUpperBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(math.pi + (1 / 11) ** 2.1, math.pi, 0),
+            math.pi
+        )
+
+    def test_saturate_justBelowUpperBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(math.pi - (1 / 100000) ** 2.43, math.pi, 0),
+            math.pi - (1 / 100000) ** 2.43
+        )
+
+    def test_saturate_justAboveLowerBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(math.pi ** 3.11, 100, math.pi ** 3.11 - (1 / 149)**16.03),
+            math.pi ** 3.11
+        )
+
+    def test_saturate_justBelowLowerBound(self):
+        self.assertEqual(
+            ControllerOutputRefiner.saturate(10, 20, 10.0000000001 - (1 / 343443) ** 2.45435),
+            10.0000000001 - (1 / 343443) ** 2.45435
+        )
+
+    # Assertion error check when lowerBound > upperBound
+    def test_saturate_badBoundInputs(self):
```

Comment:
nit picking here, can we please use "assertRaises" here: https://stackoverflow.com/questions/129507/how-do-you-test-that-a-python-function-throws-an-exception

Reply:
Good catch. I was actually getting issues with using assertRaises, so I used a try-catch block instead. I'll look into it again.

Reply:
cool, here's an example of how I used assertRaises in case it's helpful: https://github.com/UBCSailbot/boat_controller/blob/master/test/test_RudderController.py#L43

Reply:
I changed the test case. Let me know what you think.